cluster/nginx.go

```diff
@@ -140,18 +117,71 @@ func DeleteNginxLBDeployments(clientset *kubernetes.Clientset, deploymentName st
 		}
 		for _, deployment := range deployments.Items {
 			if deployment.Name != deploymentName {
-				// This delete option is to make sure that the pods belonging to
-				// the nginx-resolver deployment are 'collected' immediately.
 				fgPropagation := metav1.DeletePropagationForeground
 				fgDeleteOption := metav1.DeleteOptions{
 					PropagationPolicy: &fgPropagation,
 				}
```

Comment:
There is no need to force the deletion of the pods (and containers) belonging to the 'older' nginx-resolver deployment immediately, since we already have a new one in its place. We can use the default delete option of 'lazily' deleting the pods.